src/app/rundown-view/components/on-air-details-panel/on-air-details-panel.component.ts

```diff
@@ -1,27 +1,97 @@
 import { Tv2PieceMetadata } from '../../../core/models/tv2-piece'
 import { Tv2PieceType } from '../../../core/enums/tv2-piece-type'
 import { Part } from '../../../core/models/part'
+import { Segment } from '../../../core/models/segment'
+import { PartEntityService } from '../../../core/services/models/part-entity.service'
+import { RundownTimingContextStateService } from '../../../core/services/rundown-timing-context-state.service'
+import { Subscription } from 'rxjs'
+import { Logger } from '../../../core/abstractions/logger.service'
+import { RundownTimingContext } from '../../../core/models/rundown-timing-context'
 
 @Component({
   selector: 'sofie-on-air-details-panel',
   templateUrl: './on-air-details-panel.component.html',
   styleUrls: ['./on-air-details-panel.component.scss'],
-  changeDetection: ChangeDetectionStrategy.OnPush,
 })
-export class OnAirDetailsPanelComponent implements OnChanges {
+export class OnAirDetailsPanelComponent implements OnChanges, OnInit, OnDestroy {
   @Input()
   public rundown: Rundown
 
+  public onAirSegment?: Segment
+  public onAirPart?: Part
   public endWords?: string
+  public remainingDurationInMsForOnAirPart?: number
+  public remainingDurationInMsForOnAirSegment?: number
+  public durationInMsSpentInOnAirSegment?: number
+  private rundownTimingContextSubscription?: Subscription
+  private readonly logger: Logger
+
+  constructor(
+    private readonly partEntityService: PartEntityService,
+    private readonly rundownTimingContextStateService: RundownTimingContextStateService,
+    logger: Logger
+  ) {
+    this.logger = logger.tag('RundownSubheaderComponent')
+  }
+
+  public ngOnInit(): void {
+    this.rundownTimingContextStateService
+      .subscribeToRundownTimingContext(this.rundown.id)
+      .then(rundownTimingContextObservable => rundownTimingContextObservable.subscribe(this.onRundownTimingContextChanged.bind(this)))
+      .then(rundownTimingContextSubscription => (this.rundownTimingContextSubscription = rundownTimingContextSubscription))
+      .catch(error => this.logger.data(error).error('Failed subscribing to rundown timing context changes.'))
+  }
+
+  private onRundownTimingContextChanged(rundownTimingContext: RundownTimingContext): void {
+    this.updateOnAirPartTiming(rundownTimingContext)
+    this.updateOnAirSegmentTiming(rundownTimingContext)
+  }
+
+  private updateOnAirPartTiming(rundownTimingContext: RundownTimingContext): void {
+    if (!this.onAirPart || !this.doesPartContainVideoClipOrVoiceOver(this.onAirPart)) {
+      this.remainingDurationInMsForOnAirPart = undefined
+    } else {
+      this.remainingDurationInMsForOnAirPart = rundownTimingContext.playedDurationInMsForOnAirPart - this.partEntityService.getExpectedDuration(this.onAirPart)
+    }
+  }
+
+  private doesPartContainVideoClipOrVoiceOver(part: Part): boolean {
+    const supportedPieceTypes: (string | undefined)[] = [Tv2PieceType.VIDEO_CLIP, Tv2PieceType.VOICE_OVER]
```

Comment:
Why is the type `(string | undefined)[]`?
You are initializing the array to enum values, when are they ever undefined?

It looks like you have changed your type because of how you want to "query" the array on line 63. Same principle apply here as with naming: Call it what it is, not how you want to use it. In this case it's an array of `TvPieceType`.

Reply:
I have changed the variable type to `Tv2PieceType` and modified the associated lambda function make use of it.